### DISCUSSION TITLE
How do I recreate an ObjectBucketClaim to point to an older bucket?

### DISCUSSION THREAD

**Discussion by bh-tt** [rook/rook#11958] [Q&A]
This morning I accidentally deleted an ObjectBucketClaim. This also deleted the ObjectBucket. The ceph bucket itself was kept due to the `reclaimPolicy: Retain` on the storageClass. 

How do I create an ObjectBucket and/or ObjectBucketClaim so Rook provisions a configmap/secret that allows access to the old bucket again? Preferably using the same credentials as the old bucket but I'll settle for different credentials created by Rook.

What I have tried:

Creating a new ObjectBucketClaim without spec.generateBucketName and with spec.bucketName equal to the old bucket. This created a new ObjectBucket and credentials in a secret pointing to the same old bucket. However, I get a 403 when accessing any files in the bucket, both when using the old and when using the new credentials. I tried a `radosgw-admin bucket chown` for the new bucket and user, which did not help.

**Comment by travisn** [comment]
I would have thought setting the `bucketName` would accomplish your goal. 
@thotz What might be missing?

**Reply by thotz** [reply]
U need to create a new storageclass for Object Bucket Claims with the bucket name on it.

**Reply by bh-tt** [reply]
Thanks, together with spec.bucketName that works and even allows the same old credentials. Can I ask why creating a new storageclass is necessary?

It makes it much harder to simply reapply the same objectbucketclaim to access the same bucket. Creating a separate storageclass for each objectbucketclaim seems to be the only way to allow recreating an objectbucketclaim, which essentially means non-administrators also need to be able to create storageclasses.

**Reply by thotz** [reply]
As per the lib bucket provisioner design the existing buckets can be referred to in Storage Class, otherwise, provisioner will try to create the bucket.
